django/db/models/query.py

```diff
@@ -732,7 +737,17 @@ def values_list(self, *fields, **kwargs):
         if flat and len(fields) > 1:
             raise TypeError("'flat' is not valid when values_list is called with more than one field.")
 
-        clone = self._values(*fields)
+        _fields = []
+        expressions = {}
+        for field in fields:
+            if hasattr(field, 'resolve_expression'):
+                field_id = str(id(field))
```

Comment:
Possibility - but as discussed I can't think of a nicer way other than maybe prepending the classname to the id.